rcl/include/rcl/node_type_cache.h

```diff
@@ -0,0 +1,63 @@
+// Copyright 2023 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCL__NODE_TYPE_CACHE_H_
```

Comment:
I think this should probably be an internal API - in `src` folder. Maybe add one helper function in `node.h` - `rcl_node_get_type_description(type_hash)` for public use.

Reply:
In REP2011 it says

> There will be an option to opt-out of creating this service, and a way to start and stop the service after node creation as well.

So I guess there has to be a public API to start/stop the service?

Reply:
Sorry, my previous comment was actually meant for https://github.com/achim-k/rcl/pull/1#discussion_r1147972691, not this discussion.

About making the node cache interface private: I'm afraid that this is not possible since we call the register/unregister functions from `rcl_action/action_server.c` which is not in the rcl core `src` folder.